R/data.table.R

```diff
@@ -2684,15 +2684,16 @@ setnames = function(x,old,new,skip_absent=FALSE) {
   invisible(x)
 }
 
-setcolorder = function(x, neworder=key(x), before=NULL, after=NULL)  # before/after #4358
+setcolorder = function(x, neworder=key(x), before=NULL, after=NULL,skip_absent=FALSE)  # before/after #4358
 {
   if (is.character(neworder) && anyDuplicated(names(x)))
     stopf("x has some duplicated column name(s): %s. Please remove or rename the duplicate(s) and try again.", brackify(unique(names(x)[duplicated(names(x))])))
   if (!is.null(before) && !is.null(after))
     stopf("Provide either before= or after= but not both")
   if (length(before)>1L || length(after)>1L)
     stopf("before=/after= accept a single column name or number, not more than one")
-  neworder = colnamesInt(x, neworder, check_dups=FALSE)  # dups are now checked inside Csetcolorder below
+  neworder = colnamesInt(x, neworder, check_dups=FALSE, skip_absent=skip_absent)  # dups are now checked inside Csetcolorder below
+  neworder = neworder[!neworder %in% 0] # tests 498.11, 498.13 fail w/o this
```

Comment:
can neworder have NA?

if not, let's do neworder!=0

if so, let's do !is.na(neworder) & neworder!=0

Reply:
I don't think it can be NA but @Nj221102 should confirm.

Reply:
@sluga Yup, we can't use NA when `neworder` contains numbers (error), but you can use it when columns in `neworder` are represented by strings. This inconsistency was also mentioned in #6070, but we didn't get any use case for fixing it. However, now that we can fix it for this PR, I think we should. @MichaelChirico, what do you think?

```
dt = data.table(a=1, b=2, d=3)
> colnamesInt(dt, c("a", NA), skip_absent = TRUE)
[1] 1 0
> colnamesInt(dt, c(1, NA), skip_absent = TRUE)
Error in colnamesInt(dt, c(1, NA), skip_absent = TRUE) : 
  argument specifying columns received non-existing column(s): cols[2]=-2147483648

```

Reply:
@Nj221102 I think the question here is whether `colnamesInt()` can return `NA`

Reply:
> @Nj221102 I think the question here is whether `colnamesInt()` can return `NA`
sorry i misunderstood the question.

As far as I know, there is no way for `colnamesint` to return NA because when it doesn't find a column, it just returns 0 in its place. Additionally, NA is also converted to 0 when `cols` is character and throws an error for NA when `cols` is numeric.

Reply:
Thanks for confirming!